lib/common/common.go

```diff
@@ -177,6 +183,36 @@ func GenerateManifest(layerData types.DockerImageData, dockerURL *types.ParsedDo
 	return genManifest, nil
 }
 
+func convertVolumesToMPs(dockerVolumes map[string]struct{}) ([]appctypes.MountPoint, error) {
+	mps := []appctypes.MountPoint{}
+	dup := make(map[string]int)
+
+	for p, _ := range dockerVolumes {
+		n := filepath.Join("volume-", p)
+		sn, err := appctypes.SanitizeACName(n)
+		if err != nil {
+			return nil, err
+		}
+
+		// check for duplicate names
+		if i, ok := dup[sn]; ok {
+			dup[sn] = i + 1
+			sn = fmt.Sprintf("%s%s%d", sn, "-", i)
+		} else {
+			dup[sn] = 1
+		}
+
+		mp := appctypes.MountPoint{
```

Comment:
http://docs.docker.com/userguide/dockervolumes/#mount-a-host-directory-as-a-data-volume

"Docker defaults to a read-write volume but we can also mount a directory read-only."